### PR TITLE
Add reproducer for analysis of files added as symbolic links

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #4966

### CODE REVIEW

File: its/src/test/java/com/sonar/it/csharp/Tests.java [hunk 1]
```diff
@@ -83,6 +83,10 @@ public static BuildResult analyzeProject(Path temp, String projectDir, @Nullable
   public static BuildResult analyzeProject(String projectKey, Path temp, String projectDir, @Nullable String profileKey, String... keyValues) throws IOException {
     Path projectFullPath = TestUtils.projectDir(temp, projectDir);
 
+    return analyzeExitingProject(projectKey, projectFullPath, profileKey, keyValues);
+  }
+
+  public static BuildResult analyzeExitingProject(String projectKey, Path projectFullPath, @Nullable String profileKey, String... keyValues) {
```
Comment (pavel-mikula-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/Tests.java:89]: Typo?
```suggestion
  public static BuildResult analyzeExistingProject(String projectKey, Path projectFullPath, @Nullable String profileKey, String... keyValues) {
```

Comment (pavel-mikula-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/Tests.java:89]: What about `analyzeProjectPath`?

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SarifParserCallbackImpl.java [hunk 2]
```diff
@@ -155,8 +155,8 @@ public void onIssue(String ruleId, @Nullable String level, Location primaryLocat
       return;
     }
 
-    InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates()
-      .hasAbsolutePath(primaryLocation.getAbsolutePath()));
+    var predicate = context.fileSystem().predicates().hasAbsolutePath(primaryLocation.getAbsolutePath());
```
Comment (pavel-mikula-sonarsource) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SarifParserCallbackImpl.java:158]: Was this refactoring intentional? The PR title says "Add reproducer".

File: its/src/test/java/com/sonar/it/csharp/SymbolicLinksTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      assertThat(getComponent(sampleComponent)).isNotNull();
+      assertThat(getIssues(sampleComponent)).isNotEmpty();
+      assertThat(getComponent(linkedComponent)).isNotNull();
+
+      // reproducer: the "LinkedSample.cs" file is not part of the scanner context. Due to this, the issues for symbolic linked files are not imported.
+      assertThat(getIssues(linkedComponent)).isEmpty();
+      assertThat(buildResult.getLogs()).contains("Skipping issue S1135, input file not found or excluded: " + target.resolve("LinkedSample.cs").toAbsolutePath());
+    }
+    catch (FileSystemException exception) {
+      Assumptions.assumeTrue(false, "Symbolic links can be created only when running with administrator privileges.");
```
Comment (pavel-mikula-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/SymbolicLinksTest.java:67]: `assumeTrue(false` looks hacky. Can we just `throw new TestAbortedException` ourselves?

File: its/projects/SymbolicLinks/SymbolicLinks/SymbolicLinks.csproj [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
```
Comment (pavel-mikula-sonarsource) [inline its/projects/SymbolicLinks/SymbolicLinks/SymbolicLinks.csproj:6]: Looks like noise
```suggestion
```

File: its/projects/SymbolicLinks/Links/LinkedSample.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace SymbolicLinks;
```
Comment (pavel-mikula-sonarsource) [inline its/projects/SymbolicLinks/Links/LinkedSample.cs:1]: The link vs target confuses me a little. Can we rename the directory from `Links` to `Linked` or `LinkedFiles`.

File: its/projects/SymbolicLinks/SymbolicLinks/SymbolicLinks.csproj [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (pavel-mikula-sonarsource) [inline its/projects/SymbolicLinks/SymbolicLinks/SymbolicLinks.csproj:1]: While it's usually fine, in this case, I find `its/projects/SymbolicLinks/SymbolicLinks/SymbolicLinks.csproj` over-symbolic-linked in the name.

Can it be `its/projects/SymbolicLinks/Project/Project.csproj` or `its/projects/SymbolicLinks/WithLinks/WithLinks.csproj`, so it is easier to identify what is linking what?

While the project structure itself is fine, reading the Java code is confusing, as it has mix of both.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Some questions about renaming stuff

File: its/src/test/java/com/sonar/it/csharp/Tests.java (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/Tests.java:89]: Works :)

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**